sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using Azure.Core;
+
+namespace Azure.Data.Tables
+{
+
+    public partial class DynamicTableEntity
+    {
+        private readonly IDictionary<string, object> _properties;
+
+        /// <summary>
+        /// The partition key is a unique identifier for the partition within a given table and forms the first part of an entity's primary key.
+        /// </summary>
+        /// <value>A string containing the partition key for the entity.</value>
+        public string PartitionKey
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.PartitionKey, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.PartitionKey] = value; }
+        }
+
+        /// <summary>
+        /// The row key is a unique identifier for an entity within a given partition. Together the <see cref="PartitionKey" /> and RowKey uniquely identify every entity within a table.
+        /// </summary>
+        /// <value>A string containing the row key for the entity.</value>
+        public string RowKey
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.RowKey, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.RowKey] = value; }
+        }
+
+        /// <summary>
+        /// The Timestamp property is a DateTime value that is maintained on the server side to record the time an entity was last modified.
+        /// The Table service uses the Timestamp property internally to provide optimistic concurrency. The value of Timestamp is a monotonically increasing value,
+        /// meaning that each time the entity is modified, the value of Timestamp increases for that entity. This property should not be set on insert or update operations (the value will be ignored).
+        /// </summary>
+        /// <value>A <see cref="DateTimeOffset"/> containing the timestamp of the entity.</value>
+        public DateTimeOffset Timestamp
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.TimeStamp, out object value) ? (DateTimeOffset)value : default; }
+        }
```

Comment:
Should this be nullable?  `null` is usually a little easier to test against than an empty struct.

Reply:
In the dictionary context I don't think there should be a scenario where a property exists in the collection and is null. 

from the [docs](https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-the-table-service-data-model#property-types)
> The Table service does not persist null values for properties. When querying entities, the above property types are all non-nullable. When writing entities, the above property types are all nullable, and any property with a null value is handled as if the payload did not contain that property.

Reply:
In this particular case, what if I do `new DynamicTableEntity()`?  Checking the timestamp will be a great way to tell if I've sent something to the service.  Comparing the timestamp against null will be easier than comparing against `default(DateTimeOffset)`.

Also, what happens if I get an entity, null out one of the properties, and try to upload it?  You might argue I should have called `Dictionary.Remove` but will customers know that's meaningfully different?  I think you'll have to deal with nulls being in the dictionary for robustness even though it's not ideal from a semantic perspective.

Reply:
tracking here #13658